chia/full_node/tx_processing_queue.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+from dataclasses import dataclass
+from typing import Dict, List
+
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.transaction_queue_entry import TransactionQueueEntry
+
+
+@dataclass
+class TransactionQueue:
+    """
+    This class replaces one queue by using a high priority queue for local transactions and separate queues for peers.
+    Local transactions are processed first.
+    Then the next transaction is taken from the next non-empty queue after the last processed queue. (round-robin)
+    This decreases the effects of one peer spamming your node with transactions.
+    """
+
+    _list_iterator: int
+    _queue_length: asyncio.Semaphore
+    _index_to_peer_map: List[bytes32]
+    _queue_dict: Dict[bytes32, asyncio.Queue[TransactionQueueEntry]]
+    _high_priority_queue: asyncio.Queue[TransactionQueueEntry]
```

Comment:
I think this should be a `queue.SimpleQueue`. We don't use any of the synchronization that comes with `asyncio.Queue`

Reply:
SimpleQueue Doesn't have a peer transaction limit

Reply:
that's fine. it has a length